### PR TITLE
[IMP] im_livechat: remove AbstractThread (merge with WebsiteLivechat)

### PR DESCRIPTION
**Pull request by alexkuhn**
Task-2894477

### CODE REVIEW

File: addons/im_livechat/static/src/legacy/widgets/abstract_thread_window/abstract_thread_window.js [hunk 1]
```diff
@@ -37,11 +37,11 @@ const AbstractThreadWindow = Widget.extend({
      *
      * @abstract
      * @param {Widget} parent
-     * @param {@im_livechat/legacy/models/abstract_thread} [thread=null] the thread that this
+     * @param {@im_livechat/legacy/models/website_livechat} [thread=null] the thread that this
```
Comment (seb-odoo) [inline addons/im_livechat/static/src/legacy/widgets/abstract_thread_window/abstract_thread_window.js:40]: rename website to public

**Review by seb-odoo — APPROVED** [review]
ok if comments above resolved

File: addons/im_livechat/static/src/legacy/models/website_livechat.js [hunk 2]
```diff
@@ -263,6 +263,17 @@ const WebsiteLivechat = AbstractThread.extend(ThreadTypingMixin, {
...
+     * Post a message on this thread
+     *
+     * @abstract
+     * @private
+     * @returns {Promise} resolved with the message object to be sent to the
+     *   server
+     */
+    _postMessage() {
+        return Promise.resolve();
+    },
```
[the remark this answers is not in the document]
Comment (alexkuhn) [reply addons/im_livechat/static/src/legacy/models/website_livechat.js:276]: Yes, apparently everything was a lie... Sort of! bubbling of `message_posted` actually posts the message 🥸
All the changes are structural, and this method is still uses as async with `.then`.
I could simply code with just keyword `async`, but it's not much.

**Comment by alexkuhn** [comment]
Merged in grouped PR: https://github.com/odoo/odoo/pull/94722

**Pull request closed without merging**